File: src/server/WatchStore.Domain/Entities/CustomerAddress.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WatchStore.Domain.Entities
{
    public class CustomerAddress
    {
        public int AddressId { get; set; }  
        public int CustomerId { get; set; } 
        public string AddressLine { get; set; } // Số nhà, tên đường
        public int ProvinceId { get; set; } // Mã tỉnh/thành phố
        public string Province { get; set; }    // Tỉnh/thành phố
        public int DistrictId { get; set; } // Mã quận/huyện
        public string District { get; set; }    // Quận/huyện
        public int WardId { get; set; }     // Mã phường/xã
        public string Ward { get; set; }        // Phường/xã
        //public string ZipCode { get; set; }     // Mã bưu điện
        public bool IsDefault { get; set; } // Địa chỉ mặc định
        public DateTime CreatedAt { get; set; } = DateTime.Now;

        // Navigation Property
        public Customer Customer { get; set; }
    }
}

File: src/client/src/types/customeraddress.ts
import exp from "constants";

export type CustomerAddress = {
  addressId: number;
  customerId: number;
  addressLine: string;
  provinceId: number;
  province: string;
  district: string;
  districtId: number;
  ward: string;
  wardId: number;
  zipCode: string;
  isDefault: Boolean;
  createdAt: string;
};

export type CreateCustomerAddress = {
  customerId: number;
  addressLine: string;
  province: string;
  provinceId: number;
  district: string;
  districtId: number;
  ward: string;
  wardId: number;
  isDefault: boolean;
};

export type Province = {
  provinceID: number;
  provinceName: string;
};

export type District = {
  districtID: number;
  provinceID: number;
  districtName: string;
};

export type Ward = {
  wardCode: string;
  districtID: number;
  wardName: string;
};
